app/controllers/application_controller.rb

```diff
@@ -30,11 +30,12 @@ def authenticate_admin_user!
   end
 
   def after_sign_in_path_for(user)
-    if user.has_admin_access?
-      admin_dashboard_path
-    else
-      root_path
-    end
+    user_return_to = session.fetch(:user_return_to, '')
+    return user_return_to if user_return_to.include?(oauth_authorization_path)
+
+    return admin_dashboard_path if user.has_admin_access?
```

Comment:
I think this is a good opportunity to use a `case/when` statement to clutter those `if`'s a bit.

What do you think?

Reply:
The second statement has nothing to do with `user_return_to` value, so keeping each statement separated I think it is more legible